src/plugins/viewLargeAction/viewLargeAction.js

```diff
@@ -59,6 +59,14 @@ export default class ViewLargeAction {
     _expand(objectPath, view) {
         const element = this._getPreview(objectPath, view);
 
+        const viewContext = view.getViewContext && view.getViewContext();
+        let isPausedBeforeState;
+        const isImagery = view?.domainObject?.type === 'example.imagery';
```

Comment:
This needs to be improved before merging.

Reply:
I agree, we should try and avoid object-specific logic in this generic action. One way to approach this is to support an optional `onPreviewModeChange` event to views. Views already have the concept of optional event handlers with [the `onEditModeChanged` event handler](https://github.com/nasa/openmct/blob/1ddf5e5137508a0518f2d90bc080ffa07894016c/src/plugins/condition/ConditionSetViewProvider.js#L72). This new event handler would be called from the `viewLargeAction` when entering and leaving the preview or "view large" mode.

`ImageryView.onPreviewModeChange` could then handle the logic of pausing and unpausing, keeping it all within the view itself.

Thoughts?